tests/services/phase.service.spec.ts

```diff
@@ -1,66 +1,155 @@
     });
 
-    it('should not throw any errors if there are multiple defined openPhases', () => {
-      githubService.fetchSettingsFile.and.returnValue(of(MULTIPLE_OPEN_PHASES_SESSION_DATA));
-      phaseService.storeSessionData().subscribe({
-        error: () => fail()
-      });
+    it('should set repository if repository is valid', async () => {
+      githubServiceSpy.isRepositoryPresent.and.returnValue(of(true));
+
+      const setRepositorySpy = spyOn(phaseService, 'setRepository');
+      const repoSetSourceNext = spyOn(phaseService.repoSetSource, 'next');
+
+      await phaseService.initializeCurrentRepository();
+
+      expect(loggingServiceSpy.info).toHaveBeenCalledWith(`PhaseService: Repo is ${WATCHER_REPO}`);
+      expect(setRepositorySpy).toHaveBeenCalledWith(WATCHER_REPO);
+      expect(repoSetSourceNext).toHaveBeenCalledWith(true);
     });
 
-    it('should throw an error if no openPhases are defined', () => {
-      githubService.fetchSettingsFile.and.returnValue(of(NO_OPEN_PHASES_SESSION_DATA));
-      phaseService.storeSessionData().subscribe({
-        next: () => fail(),
-        error: (err) => expect(err).toEqual(new Error(NO_ACCESSIBLE_PHASES))
-      });
+    it('should throw error if repository is invalid', async () => {
+      githubServiceSpy.isRepositoryPresent.and.returnValue(of(false));
+
+      await expectAsync(phaseService.initializeCurrentRepository()).toBeRejectedWithError(
+        ErrorMessageService.repositoryNotPresentMessage()
+      );
     });
   });
 
-  describe('.githubRepoPermissionLevel()', () => {
-    it('should return "repo" if phaseModeration is included in openPhases', () => {
-      phaseService.sessionData = MODERATION_PHASE_SESSION_DATA;
-      expect(phaseService.sessionData.openPhases).toContain(Phase.phaseModeration);
-      expect(phaseService.githubRepoPermissionLevel()).toEqual('repo');
+  describe('changePhase(Phase)', () => {
+    beforeEach(() => {
+      githubServiceSpy = jasmine.createSpyObj('GithubService', ['storePhaseDetails']);
+      phaseService = new PhaseService(githubServiceSpy, repoUrlCacheServiceSpy, loggingServiceSpy);
     });
 
-    it('should return "public_repo" if phaseModeration is not included in openPhases', () => {
-      phaseService.sessionData = NO_OPEN_PHASES_SESSION_DATA;
-      expect(phaseService.sessionData.openPhases).not.toContain(Phase.phaseModeration);
-      expect(phaseService.githubRepoPermissionLevel()).toEqual('public_repo');
+    it('should set current phase and call storePhaseDetails of githubService', () => {
+      phaseService.setRepository(WATCHER_REPO);
+
+      expect(phaseService.currentPhase).toEqual(Phase.issuesViewer);
+
+      phaseService.changePhase(Phase.activityDashboard);
+
+      expect(phaseService.currentPhase).toEqual(Phase.activityDashboard);
+      expect(githubServiceSpy.storePhaseDetails).toHaveBeenCalledWith(WATCHER_REPO.owner, WATCHER_REPO.name);
```

Comment:
Although this method is called in the implementation, it doesn't really seem like intended behaviour of this method (in fact it seems a bit unnecessary in the actual implementation).